R/functions_netCDF.R

```diff
@@ -337,105 +424,248 @@ create_netCDF_from_array_with_variables <- function(x, locations, grid,
+  }
+
+  # vertical dimension
+  if(has_Z_verticalAxis) {
+    zdim <-  ncdf4::ncdim_def(name = vert_names,
+                              units = vert_units,
+                              vals = vert_vals)
+  }
+
+  # define dimensionality of netcdf variables -------------------------------------------------
+  var_dims <- if(isGridded) list(xdim, ydim) else list(idim)
+
+  if (has_T_timeAxis) {
     var_dims <- c(var_dims, list(tdim))
   }
 
-  var_defs <- lapply(seq_len(nl), function(k)
+  if (has_Z_verticalAxis) {
+    var_dims <- c(var_dims, list(zdim))
+  }
+
+  if(length(time_vals) > 1 || length(vert_vals) > 1) nn <- 1 else nn <- nl
+
+  var_defs <- lapply(seq_len(nn), function(k)
     ncdf4::ncvar_def(name = var_names[k], units = var_units[k],
       dim = var_dims, chunksizes = var_chunksizes, missval = NAflag,
       longname = var_longnames[k], prec = ncdf4_datatype))
 
+  # add lat and long as variables if not gridded
+  if(!isGridded){
+   latvar <-  ncdf4::ncvar_def(name = 'lat', units = 'degrees_north',
+                               dim = list(idim), chunksizes = var_chunksizes[1], missval = NAflag,
+                               longname = 'site latitiude', prec = ncdf4_datatype)
+
+   longvar <- ncdf4::ncvar_def(name = 'lon', units = 'degrees_east',
+                               dim = list(idim), chunksizes = var_chunksizes[1], missval = NAflag,
+                               longname = 'site longitude', prec = ncdf4_datatype)
+
+   var_defs <- c(var_defs, list(latvar, longvar))
+  }
+
+  # CRS defintion ------------------------------------------------------------------------
   crsdef <- ncdf4::ncvar_def(name = "crs", units = "", dim = list(),
     missval = NULL, prec = "integer")
 
-  # define dimension bounds
-  lonbnddef <- ncdf4::ncvar_def(name = "lon_bnds", units = "",
-    dim = list(bnddim, xdim), missval = NULL,
-    chunksizes = c(2L, var_chunksizes[1]))
-  latbnddef <- ncdf4::ncvar_def(name = "lat_bnds", units = "",
-    dim = list(bnddim, ydim), missval = NULL,
-    chunksizes = c(2L, var_chunksizes[2]))
+  # define dimension bands ---------------------------------------------------------------
+  if(isGridded) {
+    lonbnddef <- ncdf4::ncvar_def(name = "lon_bnds", units = "",
+                                  dim = list(bnddim, xdim), missval = NULL,
+                                  chunksizes = c(2L, var_chunksizes[1]))
 
-  if (has_time_central) {
-    tbnddef <- ncdf4::ncvar_def(name = "time_bnds", units = "",
-      dim = list(bnddim, tdim), missval = NULL, chunksizes = c(2L, 1L))
+    latbnddef <- ncdf4::ncvar_def(name = "lat_bnds", units = "",
+                                  dim = list(bnddim, ydim), missval = NULL,
+                                  chunksizes = c(2L, var_chunksizes[2]))
   }
 
-  nc_dimvars <- list(lonbnddef, latbnddef)
-  if (has_time_central) {
+  if (has_T_timeAxis) {
+      tbnddef <- ncdf4::ncvar_def(name = "time_bnds", units = "",
```

Comment:
Lines 540 and 546: Since the names of the "time" and "vertical" dimensions are user provided, shouldn't the associated bounds reflect their names accordingly? Or, we fix the "time" and "vertical" dimension names as we did for longitude/latitude.
As it is now, a user could define a "vertical" axis named "height" and the function would create associated bounds as "depth_bnds" - confusing to future users of that netCDF.